### PR TITLE
Handling tab visibility in getNextConnectionState

### PR DESCRIPTION
**Pull request by microbit-grace**
- Handle tab visibility in getNextConnectionState as Rob suggested and add logic to show radio bridge error dialog if tab is visible, otherwise show connection status as reconnecting automatically
- Consider failing twice state as a connection start over and reset `onFirstConnectAttempt` and `hasAttemptedReconnect`

Private link to related issues: 

- https://microbit-global.monday.com/boards/1550536443/pulses/1687691747
- https://microbit-global.monday.com/boards/1550536443/pulses/1696956864

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
I find this change hard to understand. Can you write a bit more explanation (on the PR in the first instance) ?

Do you think there's something different we could do in the connection library to simplify the code here / remove special cases?

**Comment by microbit-grace** [comment]
> I find this change hard to understand. Can you write a bit more explanation (on the PR in the first instance) ?

Have added more explanation in the PR description. Hope that helps!

> Do you think there's something different we could do in the connection library to simplify the code here / remove special cases?

Possibly? One way I can think of is if we introduced new connection status(es) that can distinguish between reason behind different disconnected states maybe (e.g. disconnected explicitly by user, connection loss, connect error, tab switching)? But on the other hand, I'm not sure if the connection library should be aware of these reasons if it was a general-use library

**Comment by microbit-grace** [comment]
Possibly can handle another tab visibility issue as well (private link to issue: https://microbit-global.monday.com/boards/1550536443/pulses/1696956864)

File: src/get-next-connection-state.ts [hunk 1]
```diff
@@ -59,11 +82,14 @@ export const getNextConnectionState = ({
...
     }
+    // Show reconnecting automatically when user hides browser tab and there is a connection error.
+    if (!isBrowserTabVisible) {
+      return { status: ConnectionStatus.ReconnectingAutomatically, flowType };
+    }
     if (
       // If bridge micro:bit causes radio bridge reconnect to fail twice
       hasAttempedReconnect
     ) {
-      setHasAttemptedReconnect(false);
```
Comment (microbit-grace) [inline src/get-next-connection-state.ts:66]: Letting `hasStartedOver` logic handle the resetting of this instead

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Comment by microbit-matt-hillsdon** [comment]
Thanks for working through this one.

**Pull request merged**